js/foundation-datepicker.js

```diff
@@ -1359,17 +1367,20 @@
                 return '';
             }
             var val = {
-                h: date.getUTCHours(),
+                H: date.getUTCHours(),
+                h: date.getUTCHours() % 12,
```

Comment:
@mbellet You will want to check date.getUTCHours == 0 this indicates 12:00 AM. right now when you select 12:00 am it displays as 00:00 AM. 

Possible solution is:
h: (date.getUTCHours() == 0) ? 12 : date.getUTCHours() % 12